src/datasources/system/components/SystemQueryEditor.tsx

```diff
@@ -1,32 +1,30 @@
-import React, { ChangeEvent } from 'react';
-import { InlineField, Input } from '@grafana/ui';
+import React from 'react';
+import { InlineField, InlineFieldRow, RadioButtonGroup } from '@grafana/ui';
 import { QueryEditorProps } from '@grafana/data';
 import { SystemDataSource } from '../SystemDataSource';
-import { SystemQuery } from '../types';
+import { QueryType, SystemQuery } from '../types';
 
 type Props = QueryEditorProps<SystemDataSource, SystemQuery>;
 
+const QUERY_TYPES = [
+  {label: "Metadata", value: QueryType.Metadata},
+  {label: "Summary", value: QueryType.Summary}
+]
+
 export function SystemQueryEditor({ query, onChange, onRunQuery }: Props) {
-  const onQueryTextChange = (event: ChangeEvent<HTMLInputElement>) => {
-    onChange({ ...query, queryText: event.target.value });
-  };
 
-  const onConstantChange = (event: ChangeEvent<HTMLInputElement>) => {
-    onChange({ ...query, constant: parseFloat(event.target.value) });
-    // executes the query
+  const onQueryTypeChange = (value: QueryType) => {
+    onChange({ ...query, queryClass: value })
     onRunQuery();
-  };
-
-  const { queryText, constant } = query;
+  }
 
   return (
-    <div className="gf-form">
-      <InlineField label="Constant">
-        <Input onChange={onConstantChange} value={constant} width={8} type="number" step="0.1" />
-      </InlineField>
-      <InlineField label="Query Text" labelWidth={16} tooltip="Not used yet">
-        <Input onChange={onQueryTextChange} value={queryText || ''} />
-      </InlineField>
+    <div>
+      <InlineFieldRow>
+        <InlineField label="Query Type">
+          <RadioButtonGroup options={QUERY_TYPES} onChange={onQueryTypeChange} value={query.queryClass} />
```

Comment:
Once we have that new utility function, this can be

```suggestion
          <RadioButtonGroup options={enumToOptions(QueryType)} onChange={onQueryTypeChange} value={query.queryClass} />
```